src/Metadata/AssemblyMetadata.Constructors.cs

```diff
@@ -6,6 +6,18 @@ namespace Lokad.ILPack.Metadata
 {
     internal partial class AssemblyMetadata
     {
+        public EntityHandle GetConstructorHandleConstructedType(ConstructorInfo ctor) {
```

Comment:
Why an alternative to `GetConstructorHandle`? What is the catch when should `GetConstructorHandleConstructedType` be used instead of `GetConstructorHandle`?

It feels to me that `GetConstructorHandle` should be refactored instead of introducing another methods. If there is a reason not to do that, then it needs a comment.